driver-reactive-streams/src/test/functional/com/mongodb/reactivestreams/client/syncadapter/SyncMongoIterable.java

```diff
@@ -50,6 +51,9 @@ public MongoCursor<T> cursor() {
 
     @Override
     public T first() {
+        if (wrapped instanceof BatchCursorPublisher<?>) {
```

Comment:
Is this ever false?  If not, can just cast with no check

Reply:
Ack